crates/astria-sequencer/src/transaction/fees.rs

```diff
@@ -0,0 +1,683 @@
+                },
+                current_fees.init_bridge_account_base_fee,
+                return_payment_map,
+                action_index,
+            )
+            .wrap_err("failed to increase transaction fees for init bridge account action")?,
+
+            Action::BridgeLock(act) => bridge_lock_update_fees(
+                act,
+                &mut PaymentInfo {
+                    from,
+                    to,
+                    allowed_fee_assets: &mut allowed_fee_assets,
+                    fee_asset: act.fee_asset.clone(),
+                    fees_by_asset: &mut fees_by_asset,
+                    fee_payment_map: &mut fee_payment_map,
+                },
+                current_fees.transfer_base_fee,
+                current_fees.bridge_lock_byte_cost_multiplier,
+                return_payment_map,
+                action_index,
+            )
+            .wrap_err("failed to increase transaction fees for bridge lock action")?,
+
+            Action::BridgeUnlock(act) => bridge_unlock_update_fees(
+                &mut PaymentInfo {
+                    from,
+                    to,
+                    allowed_fee_assets: &mut allowed_fee_assets,
+                    fee_asset: act.fee_asset.clone(),
+                    fees_by_asset: &mut fees_by_asset,
+                    fee_payment_map: &mut fee_payment_map,
+                },
+                current_fees.transfer_base_fee,
+                return_payment_map,
+                action_index,
+            )
+            .wrap_err("failed to increase transaction fees for bridge unlock action")?,
+
+            Action::BridgeSudoChange(act) => bridge_sudo_change_update_fees(
+                &mut PaymentInfo {
+                    from,
+                    to,
+                    allowed_fee_assets: &mut allowed_fee_assets,
+                    fee_asset: act.fee_asset.clone(),
+                    fees_by_asset: &mut fees_by_asset,
+                    fee_payment_map: &mut fee_payment_map,
+                },
+                current_fees.bridge_sudo_change_fee,
+                return_payment_map,
+                action_index,
+            )
+            .wrap_err("failed to increase transaction fees for bridge sudo change action")?,
+
+            Action::SudoAddressChange(act) => to = Some(act.new_address.address_bytes()),
+            Action::FeeAssetChange(act) => handle_fee_asset_change(act, &mut allowed_fee_assets),
+            Action::FeeChange(act) => handle_fee_change(act, &mut current_fees),
+            Action::ValidatorUpdate(_) | Action::Ibc(_) | Action::IbcRelayerChange(_) => {
+                continue;
+            }
+        }
+        action_index = action_index
+            .checked_add(1)
+            .expect("action index overflowed");
+    }
+    if return_payment_map {
+        Ok((fees_by_asset, Some(fee_payment_map)))
+    } else {
+        Ok((fees_by_asset, None))
+    }
```

Comment:
the `fee_payment_map` is generated every time no matter what `return_payment_map` is - remove the `return_payment_map`  param (also from `check_balance_and_get_fees`) and just return the payment map everytime, the caller can decide to drop or not

Reply:
nvm, i see you are using it to populate the map in the `_update_fees` functions. not really a fan of this method - i would prefer if each action had two separate functions, one for updating `fees_by_asset` and one for updating the payment map.

Reply:
I originally had it set up the way you're describing, but changed it for a couple reasons: 1) `get_and_report_tx_fees()` was over the line limit, and this was a way to cut down on line count quite significantly, and 2) some of the actions require fee calculation beyond just the base transfer fees, and this way the calculation would only need to be done once within the `_update_fees` functions and then passed to `increase_fees()`. If you think it's important to have it set up the other way, I can revert this and put a clippy allow in. Thanks for taking a look over!